src/org/zaproxy/zap/extension/proxies/ProxiesAPI.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.proxies;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+
+import net.sf.json.JSONObject;
+
+public class ProxiesAPI extends ApiImplementor {
+
+	private static final String PREFIX = "localProxies";
+
+    private static final String VIEW_ADDITIONAL_PROXIES = "additionalProxies";
+	private static final String ACTION_ADD_PROXY = "addAdditionalProxy";
+    private static final String ACTION_REMOVE_PROXY = "removeAdditionalProxy";
+	private static final String PARAM_ADDRESS = "address";
+    private static final String PARAM_PORT = "port";

```

Comment:
Seem to be some (I'm guessing copy/paste) tabs vs spaces here... L 49 to 85 too

Reply:
Ugh, thought I'd reformated all of the new files, but missed this one.